client/tests/test_updating_provider.py

```diff
@@ -0,0 +1,153 @@
+        postgres_host = "host.docker.internal" if "docker" in custom_marks else "quickstart-postgres"
+        redis_host = "host.docker.internal" if "docker" in custom_marks else "quickstart-redis"
+
+    postgres = ff.register_postgres(
+        name=postgres_name,
+        host=postgres_host,
+        port="5432",
+        database="postgres",
+        user=postgres_username,
+        password=postgres_password,
+        description=postgres_description
+    )
+
+    redis = ff.register_redis(
+        name=redis_name,
+        host=redis_host,
+        port=6379,
+        password=redis_password,
+        description=redis_description
+    )
+
+    resource_client.apply()
+
+    updated_postgres = resource_client.get_provider(postgres_name)
+    updated_redis = resource_client.get_provider(redis_name)
+
+    postgres_config_json = updated_postgres.serialized_config.decode("UTF-8")
+    postgres_config = json.loads(postgres_config_json, object_hook=lambda d: SimpleNamespace(**d))
+
+    redis_config_json = updated_redis.serialized_config.decode("UTF-8")
+    redis_config = json.loads(redis_config_json, object_hook=lambda d: SimpleNamespace(**d))
+
+    postgres_updates = [
+        updated_postgres.description == postgres_description,
+        postgres_config.Username == postgres_username,
+        postgres_config.Password == postgres_password
+    ]
+
+    redis_updates = [
+        updated_redis.description == redis_description,
+        redis_config.Password == redis_password,
+    ]
+    
+    assert all(postgres_updates) and all(redis_updates)
+
+
+@pytest.mark.parametrize(
+    "provider_source_fxt,is_local,is_insecure",
+    [
+        pytest.param("local_provider_source", True, True, marks=pytest.mark.local),
+        pytest.param("hosted_sql_provider_and_source", False, False, marks=pytest.mark.hosted),
+        pytest.param("hosted_sql_provider_and_source", False, True, marks=pytest.mark.docker),
+    ]
+)
+def test_invalid_provider_update(provider_source_fxt, is_local, is_insecure, request):
+    custom_marks = [mark.name for mark in request.node.own_markers if mark.name != 'parametrize']
+    provider, source, inference_store = request.getfixturevalue(provider_source_fxt)(custom_marks)
+
+    # Arranges the resources context following the Quickstart pattern
+    resource_client, postgres_name, redis_name = arrange_resources(provider, source, inference_store, is_local, is_insecure)
+
+    postgres = ff.register_postgres(
+        name=postgres_name,
+        host="updated-quickstart-postgres",
+        database="updated-postgres",
+        port="5432",
+        user="postgres",
+        password="password",
+        description = "A Postgres deployment we created for the Featureform quickstart"
+    )
+
+    redis = ff.register_redis(
+        name=redis_name,
+        host="updated-quickstart-redis",
+        port=6379,
+    )
+
+    with pytest.raises(ResourceRedefinedError):
```

Comment:
I dont think this will raise a resourceredefinederror. I believe it will print something like "Resource Already Exists"

Reply:
bump

Reply:
You're right. The backend returns the error, but the client swallows and prints it. Let me see how I can adapt the test to account for this. @simba-git pointed out we should really return a different type of error.